lib/Frontend/PrintingDiagnosticConsumer.cpp

```diff
@@ -112,9 +112,9 @@ void PrintingDiagnosticConsumer::handleDiagnostic(
     llvm::raw_svector_ostream Out(Text);
     DiagnosticEngine::formatDiagnosticText(Out, FormatString, FormatArgs);
   }
-  
+
   auto Msg = SM.GetMessage(Loc, SMKind, Text, Ranges, FixIts);
-  rawSM.PrintMessage(out, Msg);
+  rawSM.PrintMessage(out, Msg, ForceColors);
```

Comment:
But the actual change that made it work with `swift -Xfrontend -no-color-diagnostics` rather than `swiftc -no-color-diagnostics` was this.

Reply:
Oh, oops! Good catch.